In [11]:
import json
import copy

In [10]:
#https://bitbucket.org/frabjous/tractatus/src/master/tlp_latex.json
with open("comp_powers.json", encoding='utf-8') as f:
  comp_powers = json.load(f)
with open("clean_powers.json", encoding='utf-8') as f:
  clean_powers = json.load(f)
with open("power-template.json", encoding='utf-8') as f:
  template_power = json.load(f)

In [12]:
new_power = copy.deepcopy(template_power)

In [16]:
test_power = {
        "id": "1846",
        "name": "Windsoul Whirlwind",
        "flavor": "A twisting cyclone of rending winds forms around you, flinging foes into the air. The whirlwind moves at your command, wreaking havoc on the enemy.",
        "source": "Published in Forgotten Realms Player's Guide, page(s) 70.",
        "keywords": "Conjuration, Zone",
        "action": "Standard",
        "type": "Daily Attack",
        "level": "20",
        "class": "Whirlwind Genasi",
        "description": [
            "Hit: 3d10 + Intelligence modifier, Wisdom modifier, or Charisma modifier damage, and you slide the target 2 squares.",
            "Sustain Minor: You can move the burst 3 squares, then slide each target in the burst 1 square.",
            "Target: Each creature in the burst",
            "Attack: Intelligence + 5 vs. Fortitude, Wisdom + 5 vs. Fortitude, or Charisma + 5 vs. Fortitude Increase the bonus on attack rolls to +7 at 26th level.",
            "Effect: The zone lasts until the end of your next turn."
        ],
        "raw": "<h1 class=dailypower>Windsoul Whirlwind<span class=level>Whirlwind Genasi Attack 20</span></h1><p class=flavor><i>A twisting cyclone of rending winds forms around you, flinging foes into the air. The whirlwind moves at your command, wreaking havoc on the enemy.</i></p><p class=powerstat><b>Daily</b>   ✦     <b>Conjuration</b>, <b>Zone</b><br><b>Standard Action</b>      <b>Area</b> burst 1 within 10 squares</p><p class=powerstat><b>Target</b>: Each creature in the burst</p><p class=powerstat><b>Attack</b>: Intelligence + 5 vs. Fortitude, Wisdom + 5 vs. Fortitude, or Charisma + 5 vs. Fortitude Increase the bonus on attack rolls to +7 at 26th level.</p><p class=flavor><b>Hit</b>: 3d10 + Intelligence modifier, Wisdom modifier, or Charisma modifier damage, and you slide the target 2 squares.</p><p class=powerstat><b>Effect</b>: The zone lasts until the end of your next turn.</p><p class=flavor><b>Sustain Minor</b>: You can move the burst 3 squares, then slide each target in the burst 1 square.</p><br><p class=publishedIn>Published in Forgotten Realms Player's Guide, page(s) 70.</p>"
}

In [19]:
#print(new_power['data']['attack']['isAttack'])
comp_power = [p for p in comp_powers if p['id'] == test_power['id']]

In [20]:
new_power['name'] = test_power['name']
new_power['powerId'] = test_power['id']
new_power['data']['description']['value'] = test_power['flavor']
new_power['data']['source'] = test_power['source']
for description in test_power['description']:
    if ("Target:" in description):
        new_power['data']['target'] = description.replace("Target: ", "")
    if ("Attack:" in description):
        new_power['data']['target'] = description.replace("Target: ", "")

print(new_power)
    

{'name': 'Windsoul Whirlwind', 'type': 'power', 'data': {'description': {'value': 'A twisting cyclone of rending winds forms around you, flinging foes into the air. The whirlwind moves at your command, wreaking havoc on the enemy.', 'chat': '', 'unidentified': ''}, 'source': "Published in Forgotten Realms Player's Guide, page(s) 70.", 'activation': {'type': '', 'cost': 0, 'condition': ''}, 'duration': {'value': None, 'units': ''}, 'target': 'Each creature in the burst', 'range': {'value': None, 'long': None, 'units': ''}, 'uses': {'value': 0, 'max': 0, 'per': ''}, 'consume': {'type': '', 'target': '', 'amount': None}, 'ability': None, 'actionType': 'reaction', 'attackBonus': 0, 'chatFlavor': '', 'critical': None, 'damage': {'parts': [], 'versatile': ''}, 'damageCrit': {'parts': [], 'versatile': ''}, 'damageImp': {'parts': [], 'versatile': ''}, 'damageCritImp': {'parts': [], 'versatile': ''}, 'formula': '', 'save': {'ability': '', 'dc': None, 'scaling': 'spell'}, 'macro': {'type': 'scri

In [4]:
sections = []
for section in sectionsJson['sections']:
    try:
        newSection = {}
        newSection["label"] = section["label"]
        newSection["precision"] = section["precision"]
        newSection["fontSize"] = section["fontSize"]
        newSection["y_axis"] = section["y_axis"]
        newSection["x_axis"] = section["x_axis"]
        newSection["ger"] = " ".join(latexJson[section["label"]]["German"])
        newSection["pmc"] = " ".join(latexJson[section["label"]]["PearsMcGuinness"])
        newSection["ogd"] = " ".join(latexJson[section["label"]]["Ogden"])
        newSection["str"] = ""
        sections.append(newSection)
    except Exception as e:
        print(e)
sections.append(odd1)
sections.append(odd2)
        
tlp = {"sections": sections}
        
with open('sectionsJson-html.json', 'w', encoding='utf8') as f:
    json.dump(tlp, f, indent=4, ensure_ascii=False)

'6.0121'
'6.2311'


{
    "label": "6.0121",
    "precision": 4,
    "fontSize": "18px",
    "y_axis": 7,
    "x_axis": 111,
    "ger": "",
    "ogd": "",
    "str": ""
}

{
    "label": "6.2311",
    "precision": 4,
    "fontSize": "18px",
    "y_axis": 29,
    "x_axis": 115,
    "ger": "",
    "ogd": "",
    "str": ""
}